packages/client/hmi-client/src/temp/Clipboard.vue

```diff
@@ -0,0 +1,131 @@
+import TeraStratifiedMatrix from '@/components/model/petrinet/model-configurations/tera-stratified-matrix.vue';
+import { emptyMiraModel } from '@/model-representation/mira/mira';
+import type { Model } from '@/types/Types';
+import type {
+	MiraModel,
+	MiraTemplateParams,
+	TemplateSummary
+} from '@/model-representation/mira/mira-common';
+import { StratifiedMatrix } from '@/types/Model';
+import { updateParameter } from '@/model-representation/service';
+
+import * as sirJSON from '@/examples/strata-example.json';
+
+const model = ref<Model>();
+const mmt = ref<MiraModel>(emptyMiraModel());
+const mmtParams = ref<MiraTemplateParams>({});
+const ready = ref(false);
+
+/**
+ * Given a matrix-like CSV (with row and column labels), update the model with specified values.
+ *
+ * For example, with:
+ * ,A,B
+ * C,1,2
+ * D,3,4
+ *
+ * Will yield (A, C, 1), (A, D, 3), (B, C, 2) and (B, D, 4) update tuples
+ *
+ **/
+const updateByMatrixCSV = async (
+	model: Model,
+	paramLocationMap: Map<string, TemplateSummary[]>,
+	matrixType: string,
+	text: string
+) => {
+	// Parse
+	const rows = text.split('\n');
+	const columns = rows[0].split(',');
+
+	for (let i = 1; i < rows.length; i++) {
+		const data = rows[i].split(',');
+		for (let j = 1; j < data.length; j++) {
+			const rowLabel = data[0];
+			const colLabel = columns[j];
+
+			paramLocationMap.forEach((v, k) => {
+				if (matrixType === 'subjectControllers') {
+					if (rowLabel === v[0].subject && colLabel === v[0].controllers.join('-')) {
+						updateParameter(model, k, 'value', +data[j]);
+					}
+				} else if (matrixType === 'subjectOutcome') {
+					// TODO
+				} else if (matrixType === 'outcomeControllers') {
+					// TODO
+				}
+			});
+		}
+	}
+};
+
+const csvStringProcessor = async (item: DataTransferItem) => {
+	if (item.kind !== 'string') return;
+	if (item.type !== 'text/plain') return;
+
+	const paramLocationMap = new Map<string, TemplateSummary[]>();
+	const templateParams = Object.values(mmtParams.value);
+	templateParams.forEach((templateParam) => {
+		const params = templateParam.params;
+		params.forEach((paramName) => {
+			if (!paramLocationMap.has(paramName)) paramLocationMap.set(paramName, []);
+
+			paramLocationMap.get(paramName)?.push({
+				name: templateParam.name,
+				subject: templateParam.subject,
+				outcome: templateParam.outcome,
+				controllers: templateParam.controllers
+			});
+		});
+	});
+
+	// Presume this is a full matrix, with row/column labels
+	item.getAsString(async (text) => {
+		updateByMatrixCSV(model.value as Model, paramLocationMap, 'subjectControllers', text);
+
+		// Update
+		const response = await getMMT(model.value as Model);
+		mmt.value = response.mmt;
+		mmtParams.value = response.template_params;
+	});
+};
+
+const processPasteEvent = async (event: ClipboardEvent) => {
+	const clipboardData = event.clipboardData;
+	if (!clipboardData) return;
+	const item = clipboardData.items[0];
+	await csvStringProcessor(item);
```

Comment:
Minor:
The `csvStringProcessor` could return `item` to then allowing this `processPasteEvent` to handle the `updateByMatrix` call and `getMMT` logic  
This would take out a layer which would make reading it a bit easier and debugging a bit easier

I am not sure the plans for `csvStringProcessor` though. If it is going to be used more often than just here leave it as is 

Rather than
```
func a() 
  call b()

func c() 
  call a()
```
```
func c()
  call a()
  call b()  
```

Reply:
For some reason this does not work, it seems like part of the paste mechanism is async, though the documentation says otherwise. But I need to do everything via loops and callbacks, I can't return things cleanly.